client/src/components/RestaurantCandidateList/index.tsx

```diff
@@ -1,36 +1,87 @@
+import { useEffect, useState, useRef } from 'react';
+
 import RestaurantRow from '@components/RestaurantRow';
 import { RESTAURANT_LIST_TYPES } from '@constants/modal';
+import { useSocketStore } from '@store/socket';
+
+import { Socket } from 'socket.io-client';
 import { CandidateListModalBox, CandidateListModalLayout } from './styles';
 
-interface CandidateType {
-  id: string;
-  name: string;
-  category: string;
-  lat: number;
-  lng: number;
-  like: number;
-  rating?: number;
-  thumbnailImage?: string;
+interface CandidateType extends RestaurantType {
+  count?: number;
 }
 
 interface PropsType {
-  candidateData: RestaurantType[];
+  restaurantData: CandidateType[];
 }
 
-class CandidateMap {
-  [key: string]: CandidateType;
+interface VoteDataType {
+  restaurantId: string;
+  count: number;
+}
+interface VoteResultType {
+  message: string;
+  data?: { candidateList: VoteDataType[] };
 }
 
-export function CandidateListModal({ candidateData }: PropsType) {
+export function CandidateListModal({ restaurantData }: PropsType) {
+  const { socket } = useSocketStore((state) => state);
+  const voteList = useRef<VoteDataType[]>([]);
+  const [candidateData, setCandidateData] = useState<CandidateType[]>([]); // 투표된 음식점의 정보 데이터
+
+  if (!(socket instanceof Socket)) {
+    throw new Error();
+  }
+
+  const makeCandidateData = (candidateList: VoteDataType[]) => {
+    voteList.current = candidateList;
+    let tempList: CandidateType[] = [];
+
+    // voteList에 있는 후보 음식점들의 상세정보를 candidateData에 세팅
+    restaurantData.forEach((restaurantItem) => {
+      voteList.current.forEach((voteItem: VoteDataType) => {
+        if (restaurantItem.id === voteItem.restaurantId) {
+          // 좋아요 수 렌더링을 위해 음식점 상세정보에 투표 count값 추가
+          // eslint-disable-next-line no-param-reassign
+          restaurantItem.count = voteItem.count;
+          tempList = [...tempList, restaurantItem];
+        }
+      });
+    });
```

Comment:
```suggestion
  const makeCandidateData = (candidateList: VoteDataType[]): CandidateType[] => {
    voteList.current = candidateList;
    const tempList: CandidateType[] = [];

    // voteList에 있는 후보 음식점들의 상세정보를 candidateData에 세팅
    restaurantData.forEach((restaurantItem) => {
      voteList.current.forEach((voteItem: VoteDataType) => {
        if (restaurantItem.id === voteItem.restaurantId) {
          // 좋아요 수 렌더링을 위해 음식점 상세정보에 투표 count값 추가
          // eslint-disable-next-line no-param-reassign
          restaurantItem.count = voteItem.count;
          tempList.push(restaurantItem);
        }
      });
    });
```

계속해서 새로운 배열을 만들어 shallow copy 한 후 재할당되는 부분을 위 처럼 수정하는게 좋을 것 같습니다.